pkg/trace/api/pipeline_stats.go

```diff
@@ -0,0 +1,104 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"errors"
+	"fmt"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/fargate"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	pipelineStatsURLSuffix = "/api/v0.1/pipeline_stats"
+)
+
+// pipelineStatsEndpoint returns the pipeline intake url and the corresponding API key.
+func pipelineStatsEndpoint(cfg *config.AgentConfig) (url *url.URL, apiKey string, err error) {
+	if e := cfg.Endpoints; len(e) == 0 || e[0].Host == "" || e[0].APIKey == "" {
+		return nil, "", errors.New("config was not properly validated")
+	}
+	urlStr := cfg.Endpoints[0].Host + pipelineStatsURLSuffix
+	url, err = url.Parse(urlStr)
+	if err != nil {
+		return nil, "", fmt.Errorf("error parsing pipeline stats intake URL %s: %v", urlStr, err)
```

Comment:
```suggestion
		return nil, "", fmt.Errorf("error parsing pipeline stats intake URL %q: %v", urlStr, err)
```
Just in case it's some weird stuff. Better to have **q**uotes.